File: day01/solve.py
from pathlib import Path


def calc_calories(s):
    return (sum(int(val) for val in group.splitlines()) for group in s.split('\n\n'))


def part1(s: str):
    calories = calc_calories(s)
    return max(calories)


def part2(s: str):
    calories = calc_calories(s)
    return sum(sorted(calories)[-3:])


if __name__ == '__main__':
    DATA = Path(__file__).with_name('data.txt')

    print(part1(DATA))
    print(part2(DATA))

File: day01/solve.js
// nodejs solve.js

const input = require("fs").readFileSync("./data.txt", "utf-8").split("\n");

let sum = 0;
let val = 0;
let total = [];

for (i = 0; i < input.length; i++) {
    let line = input[i];

    val = parseInt(line)

    if (isNaN(val)) {
        total.push(sum)
        sum = 0;
    }
    else {
        sum = sum + val;
    }
}

console.log("part1", Math.max( ...total ));

total.sort((a, b) => a - b);

console.log("part2", total.at(-1) + total.at(-2) + total.at(-3))
